### ISSUE TITLE
HTML auto formatting using spaces instead of tabs

### ISSUE THREAD

**Issue by aeschli** [beautify-web/js-beautify#1551]
_From @alan-agius4 on July 31, 2018 7:36_

 

- VSCode Version: 1.25.1
- OS Version: Windows 10

Steps to Reproduce:

1. Set `"html.format.wrapAttributes": "force-aligned"` or `"html.format.wrapAttributes": force-expand-multiline` in the workspace settings
2. Set `indent_style` to tabs in `.editorconfig` for html
````js
[*.{cshtml,html}]
charset = utf-8
indent_style = tab
````
3. Format an html file.

![image](https://user-images.githubusercontent.com/17563226/43444812-03903e96-94a5-11e8-8eaf-c103647c1235.png)

![image](https://user-images.githubusercontent.com/17563226/43444880-24b8513a-94a5-11e8-9740-74b42f8a10d5.png)

I would expect that tabs are using instead of spaced.

Does this issue occur when all extensions are disabled?: Yes

_Copied from original issue: Microsoft/vscode#55423_

**Comment by aeschli** [comment]
_From @vscodebot[bot] on July 31, 2018 7:36_

(Experimental duplicate detection)
Thanks for submitting this issue. Please also check if it is already covered by an existing one, like:
- [HTML attributes - only expand multiline when long (#53187)](https://www.github.com/Microsoft/vscode/issues/53187)

**Comment by aeschli** [comment]
What is the tab/spaces setting for your editor? Check out the lower right corner.

To sync settings in `.editorconfig` with the vscode settings, check out https://marketplace.visualstudio.com/items?itemName=EditorConfig.EditorConfig

**Comment by aeschli** [comment]
_From @alan-agius4 on July 31, 2018 10:50_


Hi @aeschli my setting is `Tab Size: 4`
![image](https://user-images.githubusercontent.com/17563226/43455471-3fb47606-94c0-11e8-8f36-1f714c10f81a.png)

If I format the document, I can see that spaced are being changed to tabs but not for `wrapAttributes`, these are always spaces.


![html format vscode](https://user-images.githubusercontent.com/17563226/43455700-0a134de6-94c1-11e8-9d02-e9b63d11d86c.gif)

**Comment by aeschli** [comment]
reproduced on the https://beautifier.io/ playground:
- use tab as indent character
- set line length to 40
- custom settings:
```
{
  "wrap_attributes":   "force-aligned"
}
```
format the folloing code
```
<html>
<body>
    <div a=asdsd b=addasdas c=adadasda d=ASasS>
    </div>
</body>
</html>
```

Spaces are used to indent the attributes
![image](https://user-images.githubusercontent.com/6461412/45537617-fdaaa880-b804-11e8-9aeb-3868a977d74f.png)

**Comment by cpswaim** [comment]
I'm seeing a similar issue with wrap_attributes: force and tab indention.
Here's the setup:
# Description
> NOTE:
> * Check the list of open issues before filing a new issue.
> * Please update the expect output section to match how you would prefer the code to look.

# Input
The code looked like this before beautification:
```
// This is just a sample script. Paste
<div class="foo bar baz" data-other-tag="something else" data-another-tag="something else"></div>
```

# Current Output
The  code actually looked like this after beautification:
```
// This is just a sample script. Paste
<div class="foo bar baz"
 data-other-tag="something else"
 data-another-tag="something else"></div>
```

# Expected Output
The code should have looked like this after beautification:
```
/*Adjust the code to look how you prefer the output to be.*/
// This is just a sample script. Paste
<div class="foo bar baz"
	data-other-tag="something else"
	data-another-tag="something else">
</div>
```

## Environment
Browser User Agent:
Mozilla/5.0 (Macintosh; Intel Mac OS X 10_13_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/69.0.3497.100 Safari/537.36

Language Selected:
HTML

## Settings
Example:
```json
{
  "indent_size": "1",
  "indent_char": "\t",
  "max_preserve_newlines": "5",
  "preserve_newlines": true,
  "keep_array_indentation": false,
  "break_chained_methods": false,
  "indent_scripts": "normal",
  "brace_style": "collapse",
  "space_before_conditional": true,
  "unescape_strings": false,
  "jslint_happy": false,
  "end_with_newline": false,
  "wrap_line_length": "40",
  "indent_inner_html": false,
  "comma_first": false,
  "e4x": false,
  "wrap_attributes": "force"
}
```

In this case, should the wrap not be using a tab character rather than a space? Happy to file a new issue, if it's a separate issue from this one.

**Comment by bitwiseman** [comment]
@cpswaim You issue is even more basic. I think it is cover by #1294.

**Comment by bitwiseman** [comment]
@aeschli 
This should be fixed in 1.8.10-beta1.  See #1572.  
I'm not ready to release widely as it changes the way indenting and alignment are handled throughout the project and could have side-effects for projects that use tabs. 

Please have look and if you have people who are willing to try it out that'd be great.

**Comment by cpswaim** [comment]
Just had the opportunity to test. Manually upgraded js-beautify (to 1.9.0) in the sublime plugin I am using, and initial tests look great. Thank you!
